### PR TITLE
Logs Panel: Add CSV to download options

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What is this feature?**
Adding CSV option to logs download.
image

**Why do we need this feature?**
We got this feedback quite a bit from the logs table feedback form while it was in public preview.
While there is a CSV download in the query inspector the labels aren't extracted so it's not really helpful, also it's not anywhere near the other download options in the UI so I get that users are confused.

**Who is this feature for?**
Users of logs in explore.

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/explore/Logs/LogsMetaRow.tsx [hunk 1]
```diff
@@ -68,9 +84,22 @@ export const LogsMetaRow = React.memo(
             type: 'application/json;charset=utf-8',
           });
 
-          const fileName = `Explore-logs-${dateTimeFormat(new Date())}.json`;
           saveAs(blob, fileName);
           break;
+        case DownloadFormat.CSV:
+          const dataFrame = logRows[0].dataFrame;
```
Comment (ivanahuckova) [inline public/app/features/explore/Logs/LogsMetaRow.tsx:90]: We should process all dataFrames, not just the logRows[0].

**Review by ivanahuckova — COMMENTED** [review]
This is really good! I think we should process all `logRows` though, not just the first one - mainly with logs viz. With table only, I wouldn't mind it, as we are selecting there rows, but there we are not showing selected logs, but the first logs.

File: public/app/features/explore/Logs/LogsMetaRow.tsx (hunk 1 above)
Comment (gtk-grafana) [reply public/app/features/explore/Logs/LogsMetaRow.tsx:90]: Should we download a separate file for each dataframe?

File: public/app/features/explore/Logs/LogsMetaRow.tsx [hunk 2]
```diff
@@ -68,9 +84,22 @@ export const LogsMetaRow = React.memo(
...
+            id: 'organize',
+            options: {
+              excludeByName: {
+                ['labels']: true,
+                ['labelTypes']: true,
+              },
+            },
+          });
+          const transformedDataFrame = await lastValueFrom(transformDataFrame(transforms, [dataFrame]));
+          downloadDataFrameAsCsv(transformedDataFrame[0], fileName);
```
Comment (svennergr) [inline public/app/features/explore/Logs/LogsMetaRow.tsx:102]: Did not test, but isn't this generating a weird filename?
https://github.com/grafana/grafana/blob/main/public/app/features/inspector/utils/download.ts#L66

Comment (gtk-grafana) [reply public/app/features/explore/Logs/LogsMetaRow.tsx:102]: `Explore-logs-2024-02-15 07_32_59.json-data-2024-02-15 07_32_59.csv`
yeah lol I'll fix

**Comment by ivanahuckova** [comment]
@gtk-grafana the tests seems to be failing.

**Review by ivanahuckova — APPROVED** [review]

**Pull request merged**